test/testdrive/kafka-upsert-sources-order-by.td

```diff
@@ -0,0 +1,118 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This testdrive file is testing the order by clause with envelope upsert
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+$ kafka-create-topic topic=textbytes
+
+> CREATE SOURCE texttext
+  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT
+  VALUE FORMAT TEXT
+  INCLUDE OFFSET as o, TIMESTAMP as ts
+  ENVELOPE UPSERT ( ORDER BY ( ts ASC, o ))
+  WITH (SIZE = '1')
```

Comment:
It is best to not specify explicit size for sources. This way, the default will kick in, which is different in different CI jobs -- 4-4, 1 or 8-8. This means that any multi-worker issues will hopefully come up in one CI job or another.